Deployment Scripts for domain cluster

zhuweirong opened the pull request:
1. Deployment scripts for windows domain clusters
2. Support plain text list file - lst (so user does not have to use inf file)
3. Support password with the lst file
4. New algorithm on system saved cluster metadata
5. Fix a few issues/bugs

There're still a few TODO's
- Add a readme file with the scripts
- Include the scripts to the nuget package
- When the cluster requires passwd but user does not provide it, the exception message is not friendly.

zhuweirong commented on src/CoreLib/deploy.fs:
```diff
@@ -139,7 +139,7 @@ type DeploymentSettings() =
     static member val internal GetFirstDataDrive = DeploymentSettings.GetFirstDataDriveImpl with get, set
     /// Directory at remote node that holds Prajna operating state. 
-    static member val LocalFolder = Path.Combine(DeploymentSettings.RootFolder, "OneNet") with get
+    static member val LocalFolder = Path.Combine(DeploymentSettings.RootFolder, "Prajna") with get
```
For test clusters with existing data, on each machine, I will create a symbolic link Prajna that points to original folder

zhuweirong commented on src/CoreLib/deploy.fs:
```diff
@@ -159,7 +159,7 @@ type DeploymentSettings() =
     static member val internal ControlPort = 1081 with get, set
     /// Directory at remote node that holds Prajna data files
-    static member val DataFolder = @"OneNetData" with get
+    static member val DataFolder = @"PrajnaData" with get
```
For test clusters with existing data, on each machine, I will create a symbolic link PrajnaData that points to original folder

sanjeevm0 commented on src/CoreLib/client.fs:
```diff
@@ -225,13 +206,20 @@ type internal ClientLauncher() =
         let monTimer = ThreadPoolTimer.TimerWait ( fun _ -> sprintf "Task Queue Monitor Timer" ) listener.TaskQueue.MonitorTasks DeploymentSettings.MonitorIntervalTaskQueueInMs DeploymentSettings.MonitorIntervalTaskQueueInMs
+        
         // set authentication parameters for networking
+        Logger.Log( LogLevel.Info, (sprintf "Authentication parameters: pwd=%s keyfile=%s keyfilepwd=%s" pwd keyfile keyfilepwd ) )
```
Do we want to log passwords directly in logfile (they will be visible in the clear)?

zhuweirong replied:
No, we do not and should not. It was debug code, I will clean it.

sanjeevm0 commented on src/CoreLib/network.fs:
```diff
@@ -287,9 +287,9 @@ type [<AllowNullLiteral>] NetworkCommandQueue() as x =
-                let success = LocalDNS.BeginDNSResolve(x.MachineName, NetworkCommandQueue.EndResolveDNS, (x, port))
-                if (not success) then
-                    x.MarkFail()
+            let success = LocalDNS.BeginDNSResolve(x.MachineName, NetworkCommandQueue.EndResolveDNS, (x, port))
```
Indentation seems incorrect after else statement?

zhuweirong replied:
will fix

sanjeevm0 commented on src/CoreLib/network.fs:
```diff
@@ -307,6 +307,7 @@ type [<AllowNullLiteral>] NetworkCommandQueue() as x =
         x.RequireAuth <- requireAuth
         x.MyGuid <- myguid
         if (requireAuth) then
+            Logger.LogF(LogLevel.MildVerbose, fun _ -> sprintf "LoopbackConnect Auth: pwd=%s" pwd)
```
Same comment regarding logging of passwords in clear.

zhuweirong replied:
agree

sanjeevm0 commented on src/CoreLib/network.fs:
```diff
@@ -1340,6 +1341,12 @@ and [<AllowNullLiteral>] NetworkConnections() as x =
+    member x.InitializeAuthentication (pwd : string) : unit =
+        let keyFile =  Path.Combine( [| DeploymentSettings.LocalFolder; "Keys";  Guid.NewGuid().ToString("D") |] )
+        let keyFilePwd = Guid.NewGuid().ToString("D")
+        x.InitializeAuthentication(pwd, keyFile, String.Empty)
```
keyfilePwd cannot be generated, has to be passed in.  Plus it seems you are passing String.Empty, so perhaps it is best to just take out the line "let keyFilePwd = Guid.NewGuid().ToString("D") "

zhuweirong replied:
I found that if I leave it to empty, it does not work somehow. That's why I put something to work it around for now. I will double check.

zhuweirong committed:
Deployment scripts for windows domain clusters

1. Deployment scripts for windows domain clusters
2. Support plain text list file - lst (so user does not have to use inf file)
3. Support password with the lst file
4. New algorithm on system saved cluster metadata
5. Fix a few issues/bugs
6. Include the scripts/ClusterStatus in the nuget package
7. Add a readme
